migrations/000032_app_versions_table.up.sql

```diff
@@ -0,0 +1,489 @@
+-- with a new `app_version_id` column.
+--
+ALTER TABLE ONLY app_steps
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE app_steps
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE app_steps.app_id = app_versions.app_id
+    );
+ALTER TABLE ONLY app_steps
+    ALTER COLUMN app_version_id SET NOT NULL;
+ALTER TABLE ONLY app_steps
+    DROP COLUMN IF EXISTS app_id;
+
+--
+-- Foreign Key for `app_version_id` column in `app_steps` table.
+--
+ALTER TABLE ONLY app_steps
+    ADD CONSTRAINT app_steps_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+CREATE INDEX IF NOT EXISTS app_steps_app_version_id ON app_steps(app_version_id);
+
+--
+-- Add `app_version_id` column to the `jobs` table.
+--
+ALTER TABLE ONLY jobs
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE jobs
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE jobs.app_id = CAST(app_versions.app_id AS character varying)
+    );
+
+--
+-- Foreign Key for `app_version_id` column in `jobs` table.
+--
+ALTER TABLE ONLY jobs
+    ADD CONSTRAINT jobs_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+CREATE INDEX IF NOT EXISTS jobs_app_version_id ON jobs(app_version_id);
+
+--
+-- Add `app_version_id` column to the `quick_launches` table.
+--
+ALTER TABLE ONLY quick_launches
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE quick_launches
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE quick_launches.app_id = app_versions.app_id
+    );
+ALTER TABLE ONLY quick_launches
+    ALTER COLUMN app_version_id SET NOT NULL;
+
+--
+-- Foreign Key for `app_version_id` column in `quick_launches` table.
+--
+ALTER TABLE ONLY quick_launches
+    ADD CONSTRAINT quick_launches_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+CREATE INDEX IF NOT EXISTS quick_launches_app_version_id ON quick_launches(app_version_id);
+
+--
+-- Replace the `app_id` column in the `workflow_io_maps` table
+-- with a new `app_version_id` column.
+--
+ALTER TABLE ONLY workflow_io_maps
+    ADD COLUMN IF NOT EXISTS app_version_id uuid;
+UPDATE workflow_io_maps
+    SET app_version_id = (
+        SELECT id
+        FROM app_versions
+        WHERE workflow_io_maps.app_id = app_versions.app_id
+    );
+ALTER TABLE ONLY workflow_io_maps
+    ALTER COLUMN app_version_id SET NOT NULL;
+ALTER TABLE ONLY workflow_io_maps
+    DROP COLUMN IF EXISTS app_id;
+
+--
+-- Foreign Key for `app_version_id` column in `workflow_io_maps` table.
+--
+ALTER TABLE ONLY workflow_io_maps
+    ADD CONSTRAINT workflow_io_maps_app_version_id_fkey
+    FOREIGN KEY (app_version_id)
+    REFERENCES app_versions(id) ON DELETE CASCADE;
+
+CREATE INDEX IF NOT EXISTS workflow_io_maps_app_version_id ON workflow_io_maps(app_version_id);
```

Comment:
`workflow_io_maps` has a `UNIQUE (app_id, target_step, source_step)` which we probably need to add back in here with `app_version_id`

Reply:
Good catch 👍